backend/balancer/balancer.go

```diff
@@ -0,0 +1,94 @@
+package balancer
+
+import (
+	"context"
+	"sort"
+	"sync"
+
+	"github.com/upfluence/sql"
+)
+
+var (
+	RoundRobinBalancerBuilder   = roundRobinBalancerBulder{}
+	LeastPendingBalancerBuilder = leastPendingBalancerBuilder{}
+)
+
+type CloseFunc func(error)
+
+type BalancerBuilder interface {
+	Build([]sql.DB) Balancer
+}
+
+type Balancer interface {
+	Get(context.Context) (sql.DB, CloseFunc, error)
+}
+
+type roundRobinBalancerBulder struct{}
+
+func (roundRobinBalancerBulder) Build(dbs []sql.DB) Balancer {
+	return &roundRobinBalancer{dbs: dbs}
+}
+
+type roundRobinBalancer struct {
+	dbs []sql.DB
+
+	mu sync.Mutex
+	i  int
+}
+
+func (rrb *roundRobinBalancer) Get(context.Context) (sql.DB, CloseFunc, error) {
+	rrb.mu.Lock()
+
+	db := rrb.dbs[rrb.i]
+	rrb.i = (rrb.i + 1) % len(rrb.dbs)
+
+	rrb.mu.Unlock()
```

Comment:
Call me paranoid but I prefer to put the `Unlock` in a `defer` right after the `Lock` to avoid deadlock in case of a panic (which would only happens if someone tinker with i or pass an empty / nil DB slice)